examples/conversation_with_agent_with_finetuned_model/huggingface_model.py

```diff
@@ -0,0 +1,509 @@
+            if fine_tune_config.get("lora_config") is not None:
+                lora_config_default.update(fine_tune_config["lora_config"])
+
+        training_defaults = {
+            "per_device_train_batch_size": 1,
+            # "gradient_accumulation_steps": 1,
+            "gradient_checkpointing": False,
+            # "max_steps": 10,
+            "num_train_epochs": 10,
+            "output_dir": "./",
+            "optim": "paged_adamw_8bit",
+            "fp16": True,
+            "logging_steps": 1,
+            "learning_rate": 1e-5,
+            # "num_train_epochs": 10.0,
+        }
+
+        if fine_tune_config is not None:
+            if fine_tune_config.get("training_args") is not None:
+                training_defaults.update(fine_tune_config["training_args"])
+
+        from peft import get_peft_model
+
+        lora_config = LoraConfig(**lora_config_default)
+        model = get_peft_model(model, lora_config)
+
+        from trl import SFTTrainer
+        import transformers
+
+        def formatting_func(example: Dict[str, Any]) -> Dict[str, str]:
+            if example.get("context", "") != "":
+                input_prompt = (
+                    f"Below is an instruction that describes a task, "
+                    f"paired with an input that provides further context. "
+                    f"Write a response that appropriately "
+                    f"completes the request.\n\n"
+                    f"### Instruction:\n"
+                    f"{example['instruction']}\n\n"
+                    f"### Input: \n"
+                    f"{example['context']}\n\n"
+                    f"### Response: \n"
+                    f"{example['response']}"
+                )
+            else:
+                input_prompt = (
+                    f"Below is an instruction that describes a task. "
+                    "Write a response that appropriately "
+                    f"completes the request.\n\n"
+                    "### Instruction:\n"
+                    f"{example['instruction']}\n\n"
+                    f"### Response:\n"
+                    f"{example['response']}"
+                )
+
+            return {"text": input_prompt}
+
+        formatted_dataset = dataset.map(formatting_func)
+
+        trainer_args = transformers.TrainingArguments(**training_defaults)
+
+        trainer = SFTTrainer(
+            model,
+            train_dataset=formatted_dataset["train"],
+            eval_dataset=formatted_dataset["test"],
+            # data_collator=collator,
+            peft_config=lora_config,
+            args=trainer_args,
+            dataset_text_field="text",
+            max_seq_length=512,
+        )
+
+        logger.info(
+            "fine-tuning model",
+        )
+
+        trainer.train()
+
+        now = datetime.now()
+        time_string = now.strftime("%Y-%m-%d_%H-%M-%S")
+
+        # Specify the filename
+        log_name_temp = model.config.name_or_path.split("/")[-1]
```

Comment:
Maybe we should support users to set the saving directory.

Reply:
Added a new argument '`output_dir` for the users to customize saving directory. By default will save to the save example directory if left unspecified.